src/main/java/Parser.java

```diff
@@ -0,0 +1,75 @@
+public class Parser {
+    private int type;
+    private String cmd;
+    private Task task;
+
+    /**
+     * Creates a parser object to make sense of the user command.
+     * @param command Command given by user.
+     * @throws DukeException If command is not bye/list/done/todo/deadline/event/delete/find.
+     */
+    public Parser(String command) throws DukeException {
+        if (command.equals("bye")|command.equals("list")|command.equals("done")|command.equals("todo")
+                |command.equals("deadline")|command.equals("event")|command.equals("delete")
+                |command.equals("find")) {
+            this.cmd = command.toLowerCase();
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I dont know what that means :-(");
+        }
+    }
+
+    /**
+     * Creates a parser object to organise the details of the task provided by user.
+     * @param content Description of task given by user.
+     * @param command Command given by user.
+     * @throws DukeException If description given by user is empty.
+     */
+    public Parser(String content, Parser command) throws DukeException {
```

Comment:
It might be better to combine the two constructors together so that this constructor does not have to take in a datatype of itself as an argument.